service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/UpdateNotebookRegionStep.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.workspace.service.resource.controlled.cloud.gcp.ainotebook;
+
+import static bio.terra.workspace.common.utils.FlightUtils.getRequired;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CREATE_NOTEBOOK_REGION;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import java.util.UUID;
+
+public class UpdateNotebookRegionStep implements Step {
```

Comment:
All of the `UpdateFooRegionStep` steps seem identical -- any opportunity for reuse?